module/Activity/src/Activity/Service/Activity.php

```diff
@@ -141,7 +142,32 @@ public function createUpdateProposal(ActivityModel $oldActivity, array $params,
         $em->persist($proposal);
         $em->flush();
 
-        return $proposal;
+        if ($this->canApplyUpdateProposal($oldActivity)) {
+            $this->updateActivity($proposal);
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * Checks whether the current user is allowed to apply an update proposal for the given activity
+     *
+     * @param ActivityModel $activity
+     * @return bool indicating whether the update may be applied
+     */
+    protected function canApplyUpdateProposal(ActivityModel $activity)
+    {
+        if (!$this->isAllowed('update', $activity)) {
+            return false;
+        }
+
+        if ($this->isAllowed('approve', 'model')) {
```

Comment:
This ACL resource and action appear to be undefined. Although this works, I think it would be preferable to use a resource that is actually defined somewhere and that makes sense. Here, `$this->isAllowed('update','activity')` would work and make sense IMO.